src/test/java/org/folio/rest/impl/InvoiceLinesApiTest.java

```diff
@@ -68,11 +81,43 @@ public void deleteInvoicingInvoiceLinesByIdTest() {
 
   @Test
   public void postInvoicingInvoiceLinesTest() throws Exception {
-    String jsonBody = getMockData(INVOICE_LINE_SAMPLE_PATH);
+    logger.info("=== Test create invoice line - 201 successfully created ===");
+    InvoiceLine reqData = getMockAsJson(INVOICE_LINE_SAMPLE_PATH).mapTo(InvoiceLine.class);
+    reqData.setId(null);
+    reqData.setInvoiceLineNumber(null);
+    String body = getMockData(INVOICE_LINE_SAMPLE_PATH);
+
+    final InvoiceLine respData = verifyPostResponse(INVOICE_LINES_PATH, body, prepareHeaders(X_OKAPI_TENANT), APPLICATION_JSON, 201).as(InvoiceLine.class);
 
-    verifyPostResponse(INVOICE_LINES_PATH, jsonBody, prepareHeaders(X_OKAPI_TENANT), TEXT_PLAIN, 500);
+    String invoiceId = respData.getId();
+    String InvoiceLineNo = respData.getInvoiceLineNumber();
+
+    assertThat(invoiceId, notNullValue());
+    assertThat(InvoiceLineNo, notNullValue());
+    assertThat(MockServer.serverRqRs.get(INVOICE_LINE_NUMBER, HttpMethod.GET), hasSize(1));
+  }
+
+  @Test
+  public void testPostInvoicingInvoiceLinesWithInvoiceLineNumberGenerationFail() throws IOException {
+    logger.info("=== Test create invoice with error from storage on invoiceLineNo generation  ===");
+
+    InvoiceLine reqData = getMockAsJson(INVOICE_LINE_SAMPLE_PATH).mapTo(InvoiceLine.class);
+    reqData.setId(null);
+    reqData.setInvoiceLineNumber(null);
+    String body = getMockData(INVOICE_LINE_SAMPLE_PATH);
+    verifyPostResponse(INVOICE_LINES_PATH, body, prepareHeaders(INVOICE_LINE_NUMBER_ERROR_X_OKAPI_TENANT), APPLICATION_JSON, 500);
   }
 
+  @Test
+  public void testPostInvoiceLinesByIdLineWithoutId() throws IOException {
+    logger.info("=== Test Post Invoice Lines By Id (empty id in body) ===");
+
+    Errors resp = verifyPostResponse(INVOICE_LINES_PATH, getMockData(INVOICE_LINE_MISSING_ID__PATH),
```

Comment:
you can use the existing mock file and just remove the field, instead of creating a new file just without the id. It will be easy to maintain the mockdata files if we can reuse wherever possible

Reply:
Done. Using existing mock file.